bin/init.sh

```diff
@@ -61,104 +61,143 @@ export ISTIO_OUT=${ISTIO_OUT:-${ISTIO_BIN}}
 # Gets the download command supported by the system (currently either curl or wget)
 DOWNLOAD_COMMAND=""
 function set_download_command () {
-    # Try curl.
-    if command -v curl > /dev/null; then
-        if curl --version | grep Protocols  | grep https > /dev/null; then
-	       DOWNLOAD_COMMAND='curl -fLSs'
-	       return
-        fi
-        echo curl does not support https, will try wget for downloading files.
-    else
-        echo curl is not installed, will try wget for downloading files.
+  # Try curl.
+  if command -v curl > /dev/null; then
+    if curl --version | grep Protocols  | grep https > /dev/null; then
+      DOWNLOAD_COMMAND='curl -fLSs'
+      return
     fi
+    echo curl does not support https, will try wget for downloading files.
+  else
+    echo curl is not installed, will try wget for downloading files.
+  fi
+
+  # Try wget.
+  if command -v wget > /dev/null; then
+    DOWNLOAD_COMMAND='wget -qO -'
+    return
+  fi
+  echo wget is not installed.
+
+  echo Error: curl is not installed or does not support https, wget is not installed. \
+       Cannot download envoy. Please install wget or add support of https to curl.
+  exit 1
+}
 
-    # Try wget.
-    if command -v wget > /dev/null; then
-        DOWNLOAD_COMMAND='wget -qO -'
-        return
-    fi
-    echo wget is not installed.
+# Downloads and extract an Envoy binary if the artifact doesn't already exist.
+# Params:
+#   $1: The URL of the Envoy tar.gz to be downloaded.
+#   #2: The full path of the output binary. Must be a file under $2.
```

Comment:
Nit: `Must be a file under $2.` is no longer a requirement. Also, there is a typo `s/#2/$2/`.

Reply:
fixed